portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -4413,35 +4416,30 @@ else if (doAsGroup != null) {
 				if (liveGroup.isStaged() &&
 					!liveGroup.isStagedPortlet(portletId)) {
 
-					Layout liveGroupLayout =
-						LayoutLocalServiceUtil.fetchLayoutByUuidAndGroupId(
-							layout.getUuid(), liveGroup.getGroupId(),
-							layout.isPrivateLayout());
+					Group scopeGroup = GroupLocalServiceUtil.fetchGroup(
+						scopeGroupId);
 
-					if ((liveGroupLayout != null) &&
-						liveGroupLayout.hasScopeGroup()) {
+					if (checkStagingGroup) {
+						if (!scopeGroup.isStagingGroup() &&
+							!scopeGroup.isStagedRemotely()) {
 
-						scopeGroupId = _getScopeGroupId(
-							themeDisplay, liveGroupLayout, portletId);
-					}
-					else if (checkStagingGroup &&
-							 !liveGroup.isStagedRemotely()) {
-
-						Group stagingGroup = liveGroup.getStagingGroup();
+							Group stagingGroup = scopeGroup.getStagingGroup();
 
-						scopeGroupId = stagingGroup.getGroupId();
+							scopeGroupId = stagingGroup.getGroupId();
+						}
 					}
 					else {
-						scopeGroupId = liveGroup.getGroupId();
+						if (scopeGroup.isStagingGroup()) {
+							Group portletScopeLiveGroup =
+								scopeGroup.getLiveGroup();
+
+							scopeGroupId = portletScopeLiveGroup.getGroupId();
+						}
 					}
 				}
 			}
 		}
 
-		if (scopeGroupId <= 0) {
-			scopeGroupId = _getScopeGroupId(themeDisplay, layout, portletId);
-		}
-
 		return scopeGroupId;
 	}
```

Comment:
@MirnaGama, the fix placed `scopeGroupId = _getScopeGroupId(themeDisplay, layout, portletId);` to the beginning, this will take some unnecessary/redundant check for other cases. I think we should only do the change for the current case. We may place it to https://github.com/liferay-core-infra/liferay-portal/blob/9db4a7bf5e0c33a8aff7c761911bda5fe780a7af/portal-impl/src/com/liferay/portal/util/PortalImpl.java#L4419. And don't remove the last if block.
```
if (scopeGroupId <= 0) {
	scopeGroupId = _getScopeGroupId(themeDisplay, layout, portletId);
}
```

Reply:
Done ! https://github.com/liferay-core-infra/liferay-portal/pull/6440/commits/cbe62f59d4ea2f005b36faa7c45bedfdb1504e41